torchrec/distributed/train_pipeline.py

```diff
@@ -498,7 +498,7 @@ def _replace_fp_forward(self, model: torch.nn.Module) -> None:
                 m.forward = lambda x: x
 
     def _connect(self, dataloader_iter: Iterator[In]) -> None:
-        self._replace_fp_forward(cast(torch.nn.Module, self._model.module))
+        self._replace_fp_forward(cast(torch.nn.Module, self._model))
```

Comment:
can we move logic from L401 to init instead? 

also, what does GMP stand for?

Reply:
good point, the hack is already there (to check for DMP). Would be nice to remove it altogether, but it might break in some cases...

Reply:
I meant DMP :)